cql3/user_types.cc

```diff
@@ -162,21 +162,28 @@ void user_types::delayed_value::collect_marker_specification(shared_ptr<variable
     }
 }
 
-std::vector<cql3::raw_value> user_types::delayed_value::bind_internal(const query_options& options) {
+std::vector<bytes_opt> user_types::delayed_value::bind_internal(const query_options& options) {
     auto sf = options.get_cql_serialization_format();
-    std::vector<cql3::raw_value> buffers;
+
+    // user_types::literal::prepare makes sure that every field gets a corresponding value.
+    // For missing fields the values become nullopts.
+    assert(_type->size() == _values.size());
+
```

Comment:
What happens if we have unset fields?

Reply:
Ah, I see it is already illegal (though I don't understand why). Does it still make sense to reject unset values with non-frozen UDT?

Reply:
e.g. {a=1,b=:b,c=:c} with :b=2 and :c=unset should bind_and_get to {a=1,b=2}